### DISCUSSION TITLE
Create Nested-Type in SELECT-query

### DISCUSSION THREAD

**Discussion by inkrement** [ClickHouse/ClickHouse#33619] [Q&A] [answered]
Is it possible to create a Nested type without explicitly creating it in a create-statement? For example, I want to make one as part of an aggregation (also AggregatingMergeTree). Creating a tuple is easy. However, the columns are neither named nor do I know how to cast them correctly (especially in the context of -State/-Merge constructs).

**Comment by UnamedRus** [comment, marked as answer]
Nested is basically syntax sugar around multiple arrays.

Apart from having constant that those arrays should have the same length, it's just basic arrays.

If you look at table which you created with Nested Type, you will notice, that it doesn't have this keyword in it's schema, it's just 2 arrays.

**Reply by inkrement** [reply]
I thought this is also reflected in the underlying data structure, but this makes completely sense. Thanks for clarification!

**Reply by den-crane** [reply]
it's more complicated now https://github.com/ClickHouse/ClickHouse/issues/33498#issuecomment-1009028459 see `set flatten_nested=0;`
